### PR TITLE
Core/SmartScripts: SMART_TARGET_LOOT_RECIPIENTS

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/e2c915fb43ae27d695c0ab410fb1b14222745e27
## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -3835,6 +3835,23 @@ ObjectList* SmartScript::GetTargets(SmartScriptHolder const& e, Unit* invoker /*
                 }
                 break;
             }
+        case SMART_TARGET_LOOT_RECIPIENTS:
+            {
+                if (me)
+                {
+                    if (Group* lootGroup = me->GetLootRecipientGroup())
+                    {
+                        for (GroupReference* it = lootGroup->GetFirstMember(); it != nullptr; it = it->next())
```
Comment (Nyeriah) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:3844]: please add brackets here

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 2]
```diff
@@ -3835,6 +3835,23 @@ ObjectList* SmartScript::GetTargets(SmartScriptHolder const& e, Unit* invoker /*
...
+                            if (Player* recipient = it->GetSource())
```
Comment (Nyeriah) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:3845]: same here

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 3]
```diff
@@ -3835,6 +3835,23 @@ ObjectList* SmartScript::GetTargets(SmartScriptHolder const& e, Unit* invoker /*
...
+                                l->push_back(recipient);
+                    }
+                    else
+                    {
+                        if (Player* recipient = me->GetLootRecipient())
```
Comment (Nyeriah) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:3850]: also here

**Review by Nyeriah — CHANGES_REQUESTED** [review]
Brackets need fixing, otherwise its good to go

**Comment by IntelligentQuantum** [comment]
@Nyeriah Done

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**